sentry/src/main/java/io/sentry/SpanContext.java

```diff
@@ -94,6 +96,11 @@ public SpanContext(
     this.description = description;
     this.status = status;
     this.origin = origin;
+    final long threadId =
+        ScopesAdapter.getInstance().getOptions().getThreadChecker().currentThreadSystemId();
+    this.data = new ConcurrentHashMap<>();
+    this.data.put(SpanDataConvention.THREAD_ID, String.valueOf(threadId));
+    this.data.put(SpanDataConvention.THREAD_NAME, Thread.currentThread().getName());
```

Comment:
We're using [this lines](https://github.com/getsentry/sentry-java/pull/3725/files#diff-8bb325ca6fc9666c24826415b357d8e76a34cb455edc5709dd962bdcd34c6f16R527-R531) to set the thread name, should we also do the same thing here?

Reply:
Good call, I added `IThreadChecker.getCurrentThreadName()`, so we can reuse the logic from there